src/polling.js

```diff
@@ -0,0 +1,30 @@
+import { PollTimeout } from './errors'
+
+function poll(config) {
+  const { requestFunc, conditionChecker, remainingTime, resolve, reject } = config
+  const timeout = ((remainingTime || 0) - 25) < 0 ? remainingTime : 25
+
+  if (remainingTime <= 0) {
+    return reject(new PollTimeout('Polling timed out'))
+  }
+
+  requestFunc({ timeout })
+    .then(result => {
+      if (conditionChecker(result)) return resolve(result)
+      
+      const newRemainingTime = remainingTime - timeout
+
+      const newConfig = { 
+        requestFunc, 
+        conditionChecker, 
+        remainingTime: newRemainingTime, 
+        resolve, 
+        reject,
+      }
+      
+      return setTimeout(() => poll.call(this, newConfig), timeout)
```

Comment:
I guess the `setTimeout` is not necessary. What is your intention?

Reply:
And, if it can be removed, `poll` can return a promise object. So we could remove passing `resolve` and `reject`, right?

Reply:
Hi! The timeout was part of the previous implementation I did and you are right its no longer required. I'll make those changes.